lib/storage/metadata/mongoclient/backend.js

```diff
@@ -0,0 +1,835 @@
+/*
+ * we assume good default setting of write concern is good for all
+ * bulk writes. Note that bulk writes are not transactions but ordered
+ * writes. They may fail in between. To some extend those situations
+ * may generate orphans but not alter the proper conduct of operations
+ * (what he user wants and what we acknowledge to the user).
+ *
+ * Orphan situations may be recovered by the Lifecycle.
+ *
+ * We use proper atomic operations when needed.
+ */
+const async = require('async');
+const arsenal = require('arsenal');
```

Comment:
I'm confused, can repos have themselves as dependencies?

Reply:
This was a bad port. I brought an old file somehow.